packages/proposer/index.js

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+
+require("dotenv").config();
+const retry = require("async-retry");
+const { Logger, waitForLogger, delay } = require("@uma/financial-templates-lib");
+
+/**
+ * @notice Main runner program that executes a one-time setup module followed by a continuously polling module
+ * to propose prices. This runner can be used to run liquidators, disputers, and price proposers to the Optimistic
+ * Oracle for example.
+ * @param {Object} logger Module responsible for sending logs.
+ * @param {Function} setupFunc Async module to run once to set up for the polling module.
+ * @param {Function} pollingFunc Async module to poll continuously.
+ * @param {Number} pollingDelay The amount of seconds to wait between iterations. If set to 0 then running in serverless
+ *     mode which will exit after the loop.
+ * @param {Number} errorRetries The number of times the polling loop will re-try before throwing if an error occurs.
+ * @param {Number} errorRetriesTimeout The amount of milliseconds to wait between re-try iterations on failed loops.
+ * @param {Object} config General configuration params passed to setupFunc and pollingFunc.
+ * @return None or throws an Error.
+ */
+async function run({ logger, setupFunc, pollingFunc, pollingDelay, errorRetries, errorRetriesTimeout, config }) {
+  try {
+    // TODO: Should we set default values for the required fields in `config` and use the
```

Comment:
@UMAprotocol/eng Thoughts on defining default values for the required params in this config object that the runner script itself needs, such as `name` which is used in logging.

Reply:
I don't think that should be added here. Adding a `name` to logs, to inform the bot that reported a message in GCP or in slack is not done line this. Rather, we use the `bot-identifier` tag that is appended to the environment variables of the bot running and it extracted at the transport layer. You can see it within the Winston Logger here: https://github.com/UMAprotocol/protocol/blob/829e2c51f4ce193a58593f8f2847c73627d3f605/packages/financial-templates-lib/src/logger/Logger.js#L52

Therefore, I feel that it should be removed and we should stick to the convention where the `at` within the Winston logs defines the literal file where the log is produced and we leave the `bot-identifier` up to pulling the `name` prop from the env.

If we want to change this I think it should be done consistently through the bot stack in a separate PR.

Reply:
Ah yes I forgot about `bot-identifier`. So should I be replacing `name` with `bot-identifier` in this file?

Reply:
@chrismaree the main difference in this file is that where previously the "at" was "Liquidator#index", now there is no "Liquidator" to speak of. We could either try to use the `bot-identifier` value or use a more generic value like "Runner#index".

For example we could use:
```
const name = process.env.BOT_IDENTIFIER ? process.env.BOT_IDENTIFIER : "Runner"
```

Or maybe "Runner" would just be more accurate